macOS build failure in clangAstExporter due to missing c++17

Looks like c++17 isn't being enabled properly when building `clangAstExporter`. I tried building with `CXXFLAGS=-std=c++17`, and it appeared in the `CMAKE_CXX_FLAGS` passed to cmake, but the same error occured.
macOS 12.5.5, LLVM 16.0.1 from homebrew, x86_64 arch:
```c++
$ cargo install --git https://github.com/immunant/c2rust c2rust
    Updating git repository `https://github.com/immunant/c2rust`
  Installing c2rust v0.17.0 (https://github.com/immunant/c2rust#ff44a9f3)
    Updating crates.io index
   Compiling proc-macro2 v1.0.56
   Compiling quote v1.0.26
   Compiling unicode-ident v1.0.8
   Compiling libc v0.2.141
   Compiling syn v1.0.109
   Compiling memchr v2.5.0
   Compiling log v0.4.17
   Compiling io-lifetimes v1.0.10
   Compiling rustix v0.37.11
   Compiling thiserror v1.0.40
   Compiling serde v1.0.160
   Compiling cfg-if v1.0.0
   Compiling glob v0.3.1
   Compiling cc v1.0.79
   Compiling bitflags v1.3.2
   Compiling version_check v0.9.4
   Compiling regex-syntax v0.6.29
   Compiling minimal-lexical v0.2.1
   Compiling clang-sys v1.6.1
   Compiling aho-corasick v0.7.20
   Compiling nom v7.1.3
   Compiling libloading v0.7.4
   Compiling syn v2.0.15
   Compiling ucd-trie v0.1.5
   Compiling errno v0.3.1
   Compiling either v1.8.1
   Compiling bindgen v0.63.0
   Compiling which v4.4.0
   Compiling atty v0.2.14
   Compiling proc-macro-error-attr v1.0.4
   Compiling lazy_static v1.4.0
   Compiling termcolor v1.2.0
   Compiling print_bytes v1.2.0
   Compiling peeking_take_while v0.1.2
   Compiling regex v1.7.3
   Compiling is-terminal v0.4.7
   Compiling autocfg v1.1.0
   Compiling humantime v2.1.0
   Compiling once_cell v1.17.1
   Compiling rustc-hash v1.1.0
   Compiling shlex v1.1.0
   Compiling lazycell v1.3.0
   Compiling indexmap v1.9.3
   Compiling cexpr v0.6.0
   Compiling c2rust-build-paths v0.17.0 (/Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-build-paths)
   Compiling backtrace v0.3.67
   Compiling cmake v0.1.50
   Compiling proc-macro-error v1.0.4
   Compiling env_logger v0.10.0
   Compiling failure_derive v0.1.8
   Compiling adler v1.0.2
   Compiling heck v0.4.1
   Compiling prettyplease v0.1.25
   Compiling serde_json v1.0.96
   Compiling gimli v0.27.2
   Compiling unicode-xid v0.2.4
   Compiling rustversion v1.0.12
   Compiling time-core v0.1.0
   Compiling serde_bytes v0.11.9
   Compiling time-macros v0.2.8
   Compiling miniz_oxide v0.6.2
   Compiling thiserror-impl v1.0.40
   Compiling object v0.30.3
   Compiling itoa v1.0.6
   Compiling serde_derive v1.0.160
   Compiling half v1.8.2
   Compiling ryu v1.0.13
   Compiling synstructure v0.12.6
   Compiling pest v2.5.7
   Compiling hashbrown v0.12.3
   Compiling rustc-demangle v0.1.23
   Compiling addr2line v0.19.0
   Compiling pest_meta v2.5.7
   Compiling pest_generator v2.5.7
   Compiling time v0.3.20
   Compiling c2rust-bitfields-derive v0.17.0 (/Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-bitfields-derive)
   Compiling serde_cbor v0.11.2
   Compiling pest_derive v2.5.7
   Compiling colored v1.9.3
   Compiling arc-swap v1.6.0
   Compiling linked-hash-map v0.5.6
   Compiling anyhow v1.0.70
   Compiling os_str_bytes v6.5.0
   Compiling itertools v0.10.5
   Compiling clap_lex v0.2.4
   Compiling yaml-rust v0.4.5
   Compiling log-reroute v0.1.8
   Compiling handlebars v4.3.6
   Compiling strum_macros v0.24.3
   Compiling git-testament-derive v0.1.14
   Compiling fern v0.6.2
   Compiling c2rust-ast-exporter v0.17.0 (/Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter)
   Compiling failure v0.1.8
   Compiling clap_derive v3.2.18
   Compiling c2rust-bitfields v0.17.0 (/Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-bitfields)
   Compiling c2rust-ast-printer v0.17.0 (/Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-printer)
   Compiling c2rust-ast-builder v0.17.0 (/Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-builder)
   Compiling serde_bencode v0.2.3
   Compiling colored v2.0.0
   Compiling strsim v0.10.0
   Compiling smallvec v1.10.0
   Compiling strum v0.24.1
   Compiling no-std-compat v0.4.1
   Compiling pathdiff v0.2.1
   Compiling textwrap v0.16.0
   Compiling dtoa v1.0.6
   Compiling clap v3.2.23
   Compiling git-testament v0.2.4
   Compiling is_executable v1.0.1
error: failed to run custom build command for `c2rust-ast-exporter v0.17.0 (/Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter)`

Caused by:
  process didn't exit successfully: `/var/folders/gk/03w8k3v53t7f7gjt4pt6x3v40000gn/T/cargo-installjo9BMZ/release/build/c2rust-ast-exporter-0d18660c429f78da/build-script-build` (exit status: 101)
  --- stdout
  CMAKE_TOOLCHAIN_FILE_x86_64-apple-darwin = None
  CMAKE_TOOLCHAIN_FILE_x86_64_apple_darwin = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_x86_64-apple-darwin = None
  CMAKE_GENERATOR_x86_64_apple_darwin = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_x86_64-apple-darwin = None
  CMAKE_PREFIX_PATH_x86_64_apple_darwin = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_x86_64-apple-darwin = None
  CMAKE_x86_64_apple_darwin = None
  HOST_CMAKE = None
  CMAKE = None
  running: cd "/var/folders/gk/03w8k3v53t7f7gjt4pt6x3v40000gn/T/cargo-installjo9BMZ/release/build/c2rust-ast-exporter-290d3ce075fa11c6/out/build" && CMAKE_PREFIX_PATH="" "cmake" "/Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter/src" "-DCMAKE_OSX_ARCHITECTURES=x86_64" "-DLLVM_DIR=/usr/local/Cellar/llvm/16.0.1/lib/cmake/llvm" "-DClang_DIR=/usr/local/Cellar/llvm/16.0.1/lib/cmake/clang" "-DCMAKE_INSTALL_PREFIX=/var/folders/gk/03w8k3v53t7f7gjt4pt6x3v40000gn/T/cargo-installjo9BMZ/release/build/c2rust-ast-exporter-290d3ce075fa11c6/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64 -arch x86_64" "-DCMAKE_C_COMPILER=/usr/bin/cc" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64 -arch x86_64" "-DCMAKE_CXX_COMPILER=/usr/bin/c++" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64 -arch x86_64" "-DCMAKE_ASM_COMPILER=/usr/bin/cc" "-DCMAKE_BUILD_TYPE=Release"
  -- The C compiler identification is AppleClang 14.0.0.14000029
  -- The CXX compiler identification is AppleClang 14.0.0.14000029
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /usr/bin/cc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: /usr/bin/c++ - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Performing Test HAVE_FFI_CALL
  -- Performing Test HAVE_FFI_CALL - Success
  -- Found FFI: /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/lib/libffi.tbd  
  -- Performing Test Terminfo_LINKABLE
  -- Performing Test Terminfo_LINKABLE - Success
  -- Found Terminfo: /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/lib/libcurses.tbd  
  -- Found ZLIB: /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/lib/libz.tbd (found version "1.2.11")  
  -- Found zstd: /usr/local/lib/libzstd.dylib  
  -- Found LibXml2: /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/lib/libxml2.tbd (found version "2.9.13") 
  -- Configuring done (5.0s)
  -- Generating done (0.1s)
  -- Build files have been written to: /private/var/folders/gk/03w8k3v53t7f7gjt4pt6x3v40000gn/T/cargo-installjo9BMZ/release/build/c2rust-ast-exporter-290d3ce075fa11c6/out/build
  running: cd "/var/folders/gk/03w8k3v53t7f7gjt4pt6x3v40000gn/T/cargo-installjo9BMZ/release/build/c2rust-ast-exporter-290d3ce075fa11c6/out/build" && MAKEFLAGS="-j --jobserver-fds=14,15 --jobserver-auth=14,15" "cmake" "--build" "." "--target" "clangAstExporter" "--config" "Release"
  [  8%] Creating directories for 'tinycbor_build'
  [ 16%] Performing download step (git clone) for 'tinycbor_build'
  [ 25%] Performing update step for 'tinycbor_build'
  [ 33%] Performing patch step for 'tinycbor_build'
  patching file Makefile
  patching file Makefile.configure
  [ 41%] Performing configure step for 'tinycbor_build'
  .config file not yet created
  [ 50%] Performing build step for 'tinycbor_build'
  [ 58%] Performing install step for 'tinycbor_build'
  [ 66%] Completed 'tinycbor_build'
  [ 66%] Built target tinycbor_build
  [ 75%] Building CXX object CMakeFiles/clangAstExporter.dir/AstExporter.cpp.o
  [ 83%] Building CXX object CMakeFiles/clangAstExporter.dir/FloatingLexer.cpp.o
  [ 91%] Building CXX object CMakeFiles/clangAstExporter.dir/ExportResult.cpp.o

  --- stderr
  CMake Warning:
    Manually-specified variables were not used by the project:

      CMAKE_ASM_COMPILER
      CMAKE_ASM_FLAGS


  Cloning into 'tinycbor_build'...
  HEAD is now at d393c16 Parser: fix reading it->extra on big endian when bytesNeeded == 1
  In file included from /Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter/src/AstExporter.cpp:11:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/CommandLine.h:22:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/ArrayRef.h:12:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/Hashing.h:49:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/SwapByteOrder.h:17:
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:64:56: error: no template named 'is_integral_v' in namespace 'std'; did you mean 'is_integral'?
  template <typename T, typename = std::enable_if_t<std::is_integral_v<T>>>
                                                    ~~~~~^~~~~~~~~~~~~
                                                         is_integral
  /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/type_traits:772:29: note: 'is_integral' declared here
  struct _LIBCPP_TEMPLATE_VIS is_integral : _BoolConstant<__is_integral(_Tp)> { };
                              ^
  In file included from /Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter/src/AstExporter.cpp:11:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/CommandLine.h:22:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/ArrayRef.h:12:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/Hashing.h:49:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/SwapByteOrder.h:17:
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:64:51: error: template argument for non-type template parameter must be an expression
  template <typename T, typename = std::enable_if_t<std::is_integral_v<T>>>
                                                    ^~~~~~~~~~~~~~~~~~~~~
  /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/type_traits:550:16: note: template parameter is declared here
  template <bool _Bp, class _Tp = void> using enable_if_t = typename enable_if<_Bp, _Tp>::type;
                 ^
  In file included from /Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter/src/AstExporter.cpp:11:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/CommandLine.h:22:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/ArrayRef.h:12:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/Hashing.h:49:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/SwapByteOrder.h:17:
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:66:6: warning: constexpr if is a C++17 extension [-Wc++17-extensions]
    if constexpr (sizeof(T) == 1) {
       ^
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:68:13: warning: constexpr if is a C++17 extension [-Wc++17-extensions]
    } else if constexpr (sizeof(T) == 2) {
              ^
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:79:13: warning: constexpr if is a C++17 extension [-Wc++17-extensions]
    } else if constexpr (sizeof(T) == 4) {
              ^
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:92:13: warning: constexpr if is a C++17 extension [-Wc++17-extensions]
    } else if constexpr (sizeof(T) == 8) {
              ^
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:109:56: error: no template named 'is_unsigned_v' in namespace 'std'; did you mean 'is_unsigned'?
  template <typename T, typename = std::enable_if_t<std::is_unsigned_v<T>>>
                                                    ~~~~~^~~~~~~~~~~~~
                                                         is_unsigned
  /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/type_traits:1515:50: note: 'is_unsigned' declared here
  template <class _Tp> struct _LIBCPP_TEMPLATE_VIS is_unsigned : public __libcpp_is_unsigned<_Tp> {};
                                                   ^
  In file included from /Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter/src/AstExporter.cpp:11:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/CommandLine.h:22:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/ArrayRef.h:12:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/Hashing.h:49:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/SwapByteOrder.h:17:
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:109:51: error: template argument for non-type template parameter must be an expression
  template <typename T, typename = std::enable_if_t<std::is_unsigned_v<T>>>
                                                    ^~~~~~~~~~~~~~~~~~~~~
  /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/type_traits:550:16: note: template parameter is declared here
  template <bool _Bp, class _Tp = void> using enable_if_t = typename enable_if<_Bp, _Tp>::type;
                 ^
  In file included from /Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter/src/AstExporter.cpp:11:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/CommandLine.h:22:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/ArrayRef.h:12:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/Hashing.h:49:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/SwapByteOrder.h:17:
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:180:22: error: no template named 'is_unsigned_v' in namespace 'std'; did you mean 'is_unsigned'?
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~^~~~~~~~~~~~~
                       is_unsigned
  /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/type_traits:1515:50: note: 'is_unsigned' declared here
  template <class _Tp> struct _LIBCPP_TEMPLATE_VIS is_unsigned : public __libcpp_is_unsigned<_Tp> {};
                                                   ^
  In file included from /Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter/src/AstExporter.cpp:11:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/CommandLine.h:22:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/ArrayRef.h:12:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/Hashing.h:49:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/SwapByteOrder.h:17:
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:180:38: error: expected '(' for function-style cast or type construction
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~~~~~~~~~~~~~~~~~^
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:246:22: error: no template named 'is_unsigned_v' in namespace 'std'; did you mean 'is_unsigned'?
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~^~~~~~~~~~~~~
                       is_unsigned
  /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/type_traits:1515:50: note: 'is_unsigned' declared here
  template <class _Tp> struct _LIBCPP_TEMPLATE_VIS is_unsigned : public __libcpp_is_unsigned<_Tp> {};
                                                   ^
  In file included from /Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter/src/AstExporter.cpp:11:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/CommandLine.h:22:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/ArrayRef.h:12:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/Hashing.h:49:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/SwapByteOrder.h:17:
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:246:38: error: expected '(' for function-style cast or type construction
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~~~~~~~~~~~~~~~~~^
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:259:22: error: no template named 'is_unsigned_v' in namespace 'std'; did you mean 'is_unsigned'?
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~^~~~~~~~~~~~~
                       is_unsigned
  /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/type_traits:1515:50: note: 'is_unsigned' declared here
  template <class _Tp> struct _LIBCPP_TEMPLATE_VIS is_unsigned : public __libcpp_is_unsigned<_Tp> {};
                                                   ^
  In file included from /Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter/src/AstExporter.cpp:11:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/CommandLine.h:22:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/ArrayRef.h:12:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/Hashing.h:49:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/SwapByteOrder.h:17:
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:259:38: error: expected '(' for function-style cast or type construction
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~~~~~~~~~~~~~~~~~^
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:272:22: error: no template named 'is_unsigned_v' in namespace 'std'; did you mean 'is_unsigned'?
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~^~~~~~~~~~~~~
                       is_unsigned
  /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/type_traits:1515:50: note: 'is_unsigned' declared here
  template <class _Tp> struct _LIBCPP_TEMPLATE_VIS is_unsigned : public __libcpp_is_unsigned<_Tp> {};
                                                   ^
  In file included from /Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter/src/AstExporter.cpp:11:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/CommandLine.h:22:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/ArrayRef.h:12:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/Hashing.h:49:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/SwapByteOrder.h:17:
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:272:38: error: expected '(' for function-style cast or type construction
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~~~~~~~~~~~~~~~~~^
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:282:22: error: no template named 'is_unsigned_v' in namespace 'std'; did you mean 'is_unsigned'?
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~^~~~~~~~~~~~~
                       is_unsigned
  /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/type_traits:1515:50: note: 'is_unsigned' declared here
  template <class _Tp> struct _LIBCPP_TEMPLATE_VIS is_unsigned : public __libcpp_is_unsigned<_Tp> {};
                                                   ^
  In file included from /Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter/src/AstExporter.cpp:11:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/CommandLine.h:22:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/ArrayRef.h:12:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/Hashing.h:49:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/SwapByteOrder.h:17:
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:282:38: error: expected '(' for function-style cast or type construction
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~~~~~~~~~~~~~~~~~^
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:292:22: error: no template named 'is_unsigned_v' in namespace 'std'; did you mean 'is_unsigned'?
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~^~~~~~~~~~~~~
                       is_unsigned
  /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/type_traits:1515:50: note: 'is_unsigned' declared here
  template <class _Tp> struct _LIBCPP_TEMPLATE_VIS is_unsigned : public __libcpp_is_unsigned<_Tp> {};
                                                   ^
  In file included from /Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter/src/AstExporter.cpp:11:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/CommandLine.h:22:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/ArrayRef.h:12:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/Hashing.h:49:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/SwapByteOrder.h:17:
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:292:38: error: expected '(' for function-style cast or type construction
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~~~~~~~~~~~~~~~~~^
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:307:22: error: no template named 'is_unsigned_v' in namespace 'std'; did you mean 'is_unsigned'?
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~^~~~~~~~~~~~~
                       is_unsigned
  /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/type_traits:1515:50: note: 'is_unsigned' declared here
  template <class _Tp> struct _LIBCPP_TEMPLATE_VIS is_unsigned : public __libcpp_is_unsigned<_Tp> {};
                                                   ^
  In file included from /Users/melvyn/.cargo/git/checkouts/c2rust-2a3ef4c32beab81a/ff44a9f/c2rust-ast-exporter/src/AstExporter.cpp:11:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/CommandLine.h:22:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/ArrayRef.h:12:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/Hashing.h:49:
  In file included from /usr/local/Cellar/llvm/16.0.1/include/llvm/Support/SwapByteOrder.h:17:
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:307:38: error: expected '(' for function-style cast or type construction
    static_assert(std::is_unsigned_v<T>,
                  ~~~~~~~~~~~~~~~~~~~~~^
  /usr/local/Cellar/llvm/16.0.1/include/llvm/ADT/bit.h:348:56: error: no template named 'is_unsigned_v' in namespace 'std'; did you mean 'is_unsigned'?
  template <typename T, typename = std::enable_if_t<std::is_unsigned_v<T>>>
                                                    ~~~~~^~~~~~~~~~~~~
                                                         is_unsigned
  /Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include/c++/v1/type_traits:1515:50: note: 'is_unsigned' declared here
  template <class _Tp> struct _LIBCPP_TEMPLATE_VIS is_unsigned : public __libcpp_is_unsigned<_Tp> {};
                                                   ^
  fatal error: too many errors emitted, stopping now [-ferror-limit=]
  4 warnings and 20 errors generated.
  gmake[3]: *** [CMakeFiles/clangAstExporter.dir/build.make:76: CMakeFiles/clangAstExporter.dir/AstExporter.cpp.o] Error 1
  gmake[2]: *** [CMakeFiles/Makefile2:274: CMakeFiles/clangAstExporter.dir/all] Error 2
  gmake[1]: *** [CMakeFiles/Makefile2:281: CMakeFiles/clangAstExporter.dir/rule] Error 2
  gmake: *** [Makefile:215: clangAstExporter] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2

  build script failed, must exit now', /Users/melvyn/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cmake-0.1.50/src/lib.rs:1098:5
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: failed to compile `c2rust v0.17.0 (https://github.com/immunant/c2rust#ff44a9f3)`, intermediate artifacts can be found at `/var/folders/gk/03w8k3v53t7f7gjt4pt6x3v40000gn/T/cargo-installjo9BMZ`
```

--- a/c2rust-ast-exporter/src/CMakeLists.txt
+++ b/c2rust-ast-exporter/src/CMakeLists.txt
@@ -26,7 +26,7 @@ ExternalProject_Add(tinycbor_build
             PATCH_COMMAND patch --forward -p1 < ${CMAKE_CURRENT_SOURCE_DIR}/tinycbor_fix_build.patch || true
             CONFIGURE_COMMAND make .config && cat ${CMAKE_CURRENT_SOURCE_DIR}/tinycbor.config >> .config
             BUILD_COMMAND make --quiet prefix=<INSTALL_DIR> CFLAGS=-fPIC
-            INSTALL_COMMAND make --quiet prefix=<INSTALL_DIR> install            
+            INSTALL_COMMAND make --quiet prefix=<INSTALL_DIR> install
             BUILD_IN_SOURCE 1
             BUILD_BYPRODUCTS ${CMAKE_BINARY_DIR}/lib/libtinycbor.a
 )
@@ -98,7 +98,7 @@ target_link_libraries(c2rust-ast-exporter PRIVATE
   )
 
 set_target_properties(clangAstExporter PROPERTIES
-  CXX_STANDARD 14
+  CXX_STANDARD 17 # will decay to 14 if compiler doesn't support c++17
   CXX_EXTENSIONS OFF
   )
 target_link_libraries(clangAstExporter PRIVATE

--- a/scripts/test_translator.py
+++ b/scripts/test_translator.py
@@ -233,6 +233,10 @@ class TestDirectory:
         # set self.target to a known-working target tuple for it
         self.target = None
 
+        # include the compiler resource directory in compile_commands.json
+        _, stdout, _ = clang["-print-resource-dir"].run(retcode=None)
+        self.clang_resource_dir = " \"-I{}/include\",".format(stdout.strip())
+
         # parse target arch from directory name if it includes a dot
         split_by_dots = self.name.split('.')
         if len(split_by_dots) > 1:
@@ -319,12 +323,12 @@ class TestDirectory:
         compile_commands = """ \
         [
           {{
-            "arguments": [ "cc", "-D_FORTIFY_SOURCE=0", "-c", {2}"{0}" ],
+            "arguments": [ "cc", "-D_FORTIFY_SOURCE=0",{3} "-c", {2}"{0}" ],
             "directory": "{1}",
             "file": "{0}"
           }}
         ]
-        """.format(cfile, directory, target_args)
+        """.format(cfile, directory, target_args, self.clang_resource_dir)
 
         cc_db = os.path.join(directory, "compile_commands.json")
 

